smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -53,12 +53,25 @@ class MiddlewareExecutionGenerator(
             )
         } else {
             writer.write(
-                "let builder = OrchestratorBuilder<\$N, \$N, \$N, \$N, \$N>()",
+                "let builder = \$N<\$N, \$N, \$N, \$N, \$N>()",
+                ClientRuntimeTypes.Operation.OrchestratorBuilder,
                 inputShape,
                 outputShape,
                 ClientRuntimeTypes.Http.SdkHttpRequest,
                 ClientRuntimeTypes.Http.HttpResponse,
-                ClientRuntimeTypes.Http.HttpContext
+                ClientRuntimeTypes.Http.HttpContext,
+            )
+            writer.write("config.interceptorProviders.forEach { builder.interceptors.add($$0.create()) }")
+            // Swift can't infer the generic arguments to `create` for some reason
+            writer.write(
+                """
+                config.httpInterceptorProviders.forEach {
+                    let i: any HttpInterceptor<${'$'}N, ${'$'}N> = $$0.create()
+                    builder.interceptors.add(i)
+                }
```

Comment:
same nit here might be better to use a name instead of `$0.create` for readability